src/sidebar/helpers/mentions.ts

```diff
@@ -96,27 +96,53 @@ export function wrapDisplayNameMentions(
 // some assumptions about the tag content in order to be able to get away
 // with it. Specifically we assume that these tags will not contain `<` or
 // `>` in attribute values or the HTML content of the tag.
-const MENTION_TAG_RE = /<a[^>]\bdata-hyp-mention\b[^>]*>([^<]+)<\/a>/g;
+const MENTION_TAG_RE = /<a[^>]\bdata-hyp-mention\b[^>]*>@([^<]+)<\/a>/g;
+
+export type UnwrapMentionsOptions = {
+  text: string;
+  mentionMode: MentionMode;
+  mentions?: Mention[];
+};
 
 /**
  * Replace all mentions wrapped in the special `<a data-hyp-mention />` tag with
  * their plain-text representation.
+ *
  * The plain-text representation depends on the mention mode:
  * - `username`: @username
  * - `display-name`: @[Display Name]
+ *
+ * If a list of mentions is provided, the tag's userid will be matched against
+ * it, so that the plain-text version uses the most recent username or display
+ * name.
```

Comment:
I think it would be helpful here to spell out when this matters: when the user has been renamed or had their display name changed since the mention was initially created.